internal/core/application/trade_service.go

```diff
@@ -762,20 +764,20 @@ func getBalanceByAsset(unspents []domain.Unspent) map[string]uint64 {
 // depending on the trade type and the base asset amount provided.
 // The market fees are either added or subtracted to the converted amount
 // basing on the trade type.
-func calcPreviewAmount(market *domain.Market, tradeType int, amount uint64) uint64 {
+func calcPreviewAmount(market *domain.Market, tradeType int, amount uint64, asset string) uint64 {
+	var price decimal.Decimal
+	if asset == market.BaseAsset {
+		price = market.QuoteAssetPrice()
+	} else {
+		price = market.BaseAssetPrice()
+	}
+
+	chargeFeesOnTheWayIn := asset == market.BaseAsset
```

Comment:
mmm i donth think chargeFeesOnTheWayIn means this

Reply:
do you have any suggestion on how to rename this? I kept the same nomenclature of the marketmaking package since calcProposeAmount is the equivalent of the inGivenOut, while calcExpectedAmount corresponds to OutGivenIn if we try to make a comparison between strategy methods.

Also, I'm thinking that maybe these functions could be moved to the market domain in the future, wrapped by one method that takes care of calculating the price depending on its current strategy. What do you think? Eventually, this won't be done here but in another PR, after opening a dedicated ticket.

Reply:
in the sens we should change the name of the variable to not make confusuon

Reply:
The idea od being a pkg is that other people can make their own and we use them, as long they implement the same interface.

About the name I guess the coorect way is : "isBaseAsset" beaccuse is what is actually is

Reply:
I'm saying that, for now, we decide at the application level if the preview amount is calculated from a formula (balance strategy), or from a price (pluggable strategy). I think that it would be fine if we had a SpotPrice method in the market domain so that we move this logic to the domain level.